Formula/firefoxpwa.rb

```diff
@@ -0,0 +1,61 @@
+class Firefoxpwa < Formula
+  desc "Tool to install, manage and use Progressive Web Apps in Mozilla Firefox"
+  homepage "https://github.com/filips123/FirefoxPWA"
+  url "https://github.com/filips123/FirefoxPWA/archive/refs/tags/v1.0.0.tar.gz"
+  sha256 "47cb03a8f5773da235e360655a4ef93203d7d3ed760ebf6013692ec20239c2c1"
+  license "MPL-2.0"
+  head "https://github.com/filips123/FirefoxPWA.git"
+
+  depends_on "rust" => :build
+
+  def install
+    cd "native"
+
+    # Prepare the project to work with Homebrew
+    ENV["FFPWA_EXECUTABLES"] = bin
+    ENV["FFPWA_SYSDATA"] = share
+    system "bash", "./packages/brew/configure.sh", version, bin, libexec
+
+    # Build and install the project
+    system "cargo", "install", *std_cargo_args
+
+    # Install all files
+    libexec.install bin/"firefoxpwa-connector"
+    share.install "manifests/brew.json" => "firefoxpwa.json"
+    share.install "userchrome/"
+    bash_completion.install "target/release/completions/firefoxpwa.bash" => "firefoxpwa"
+    fish_completion.install "target/release/completions/firefoxpwa.fish"
+    zsh_completion.install "target/release/completions/_firefoxpwa"
+  end
+
+  def caveats
+    filename = "firefoxpwa.json"
+
+    source = share
+    destination = "/Library/Application Support/Mozilla/NativeMessagingHosts"
+
+    on_linux do
+      destination = "/usr/lib/mozilla/native-messaging-hosts"
+    end
+
+    <<~EOS
+      Before you can use the browser extension, you will need to manually link the app manifest
+      This is needed because Homebrew formulae cannot access directories outside Homebrew directory
+
+      To link the manifest, run the following commands after the formula is installed:
+      $ sudo mkdir -p #{destination}
+      $ sudo ln -sf "#{source}/#{filename}" "#{destination}/#{filename}"
+
+      #{Tty.red}You will not be able to use the extension until you link the manifest!#{Tty.reset}
```

Comment:
```suggestion
      To use the browser extension, manually link the app manifest with:
        sudo mkdir -p #{destination}
        sudo ln -sf "#{source}/#{filename}" "#{destination}/#{filename}"
```
Let's make this caveat less verbose.